moveit_core/robot_model/include/moveit/robot_model/robot_model.hpp

```diff
@@ -265,7 +268,13 @@ class RobotModel
    * what you went for. Instead, updateStateWithLinkAt(getRigidlyConnectedParentLinkModel(grasp_frame), ...)
    * will actually warp wrist (and all its descendants).
    */
-  static const moveit::core::LinkModel* getRigidlyConnectedParentLinkModel(const LinkModel* link);
+  static const LinkModel* getRigidlyConnectedParentLinkModel(const LinkModel* link, Eigen::Isometry3d& transform,
```

Comment:
Is the transform intended to be some implicit output? I don't like this API, especially given that there is no documentation around this provided. I think we need to add documentation of the parameters of this function and if a reference is passed as non-const to a function in order to be modified by it, I think it should be the last argument so that they are ordered, first inputs and second output argument.